compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -234,6 +234,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             ) => self.check_expr_path(qpath, expr, args),
             _ => self.check_expr_kind(expr, expected),
         });
+        let ty = self.resolve_vars_if_possible(ty);
```

Comment:
super tiny nit: Not sure why this commit title mentions "let chain `ty`"s -- this resolves all checked expr types :laughing: